### PR TITLE
#147 - names normalization in proxy

### LINKED ISSUE #147: Save content to cache by normalized key
We should save content by normalized keys in proxy cache to avoid caching of the same content twice. For example `https://pypi.org/simple` redirects `/AlarmTime` to `/alarmtime`, so saving content by normalized keys we can provide more reliable proxy.

### PR DESCRIPTION
**Pull request by olenagerasimova**
Closes #147 
Added names normalization in proxy and corresponding test.

### CODE REVIEW

File: src/test/java/com/artipie/pypi/http/ProxySliceTest.java [hunk 1]
```diff
@@ -142,4 +142,32 @@ void proxiesStatusAndHeaderFromRemoteOnError() {
...
 
+    @ParameterizedTest
+    @CsvSource({
+        "my project versions list in html,My_Project,my-project",
+        "my project wheel,My.Project.whl,My.Project.whl",
+        "John's project tar,Johns.Project.tar.gz,Johns.Project.tar.gz",
+        "Another index,AnotherIndex,anotherindex"
+    })
+    void normalisesNamesWhenNecessary(final String data, final String line, final String key) {
+        final byte[] body = data.getBytes();
```
Comment (olegmoz) [inline src/test/java/com/artipie/pypi/http/ProxySliceTest.java:153]: @olenagerasimova is it really needed to have different `data` in each test case? I'd suggest to simplify the test a little bit and use same data in all cases

**Review by olegmoz — CHANGES_REQUESTED** [review]
@olenagerasimova thanks, please see one comment

**Comment by olenagerasimova** [comment]
@olegmoz thanks, corrected

**Review by olegmoz — APPROVED** [review]
@olenagerasimova thanks, approved

**Comment by 0crat** [comment]
Job `gh:artipie/pypi-adapter#148` is not assigned, can't get performer

**Pull request merged**